src/Features/CSharp/Portable/ExtractInterface/CSharpExtractInterfaceService.cs

```diff
@@ -119,5 +135,45 @@ internal override bool ShouldIncludeAccessibilityModifier(SyntaxNode typeNode)
             var typeDeclaration = typeNode as TypeDeclarationSyntax;
             return typeDeclaration.Modifiers.Any(m => SyntaxFacts.IsAccessibilityModifier(m.Kind()));
         }
+
+        internal override Solution GetSolutionWithSameFileUpdated(
+            Solution solution,
+            INamedTypeSymbol extractedInterfaceSymbol,
+            IEnumerable<ISymbol> includedMembers,
+            Dictionary<ISymbol, SyntaxAnnotation> symbolToDeclarationAnnotationMap,
+            List<DocumentId> documentIds,
+            SyntaxAnnotation typeNodeAnnotation,
+            DocumentId documentIdWithTypeNode,
+            CancellationToken cancellationToken)
+        {
+            var document = solution.GetDocument(documentIdWithTypeNode);
+            var originalRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+            var typeDeclaration = originalRoot.GetAnnotatedNodes<TypeDeclarationSyntax>(typeNodeAnnotation).Single();
+
+            document = document.WithSyntaxRoot(originalRoot.TrackNodes(typeDeclaration));
+
+            var currentRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+            var editor = new SyntaxEditor(currentRoot, CSharpSyntaxGenerator.Instance);
+
+            typeDeclaration = currentRoot.GetCurrentNode(typeDeclaration);
+
+            var interfaceTypeSyntax = GetTypeSyntaxFromNamedSymbol(extractedInterfaceSymbol);
+
+            var languageService = solution.Workspace.Services.GetLanguageServices(LanguageNames.CSharp);
+            var codeGenService = languageService.GetService<ICodeGenerationService>();
+            var interfaceNode = codeGenService.CreateNamedTypeDeclaration(extractedInterfaceSymbol).WithAdditionalAnnotations(SimplificationHelpers.SimplifyModuleNameAnnotation);
```

Comment:
it seems weird to have this code here.  Doesn't the previous codepath have to generate this as well?  I would expect to see this all calling into common code.